sdk/storage/Azure.Storage.Files.DataLake/src/PathClient.cs

```diff
@@ -0,0 +1,1462 @@
+        public virtual async Task<Response<PathClient>> RenameAsync(
+            string destinationPath,
+            DataLakeRequestConditions destConditions = default,
+            DataLakeRequestConditions sourceConditions = default,
+            CancellationToken cancellationToken = default) =>
+            await RenameInternal(
+                destinationPath,
+                destConditions,
+                sourceConditions,
+                true, // async
+                cancellationToken)
+                .ConfigureAwait(false);
+
+        /// <summary>
+        /// The <see cref="RenameInternal"/> operation renames a file or directory.
+        ///
+        /// For more information, see https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create.
+        /// </summary>
+        /// <param name="destinationPath">
+        /// The destination path to rename the path to.
+        /// </param>
+        /// <param name="destConditions">
+        /// Optional <see cref="DataLakeRequestConditions"/> to add
+        /// conditions on the creation of this file or directory.
+        /// </param>
+        /// <param name="sourceConditions">
+        /// Optional <see cref="DataLakeRequestConditions"/> to add
+        /// conditions on the source on the creation of this file or directory.
+        /// </param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{PathInfo}"/> describing the
+        /// newly created page blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        private async Task<Response<PathClient>> RenameInternal(
+            string destinationPath,
+            DataLakeRequestConditions destConditions,
+            DataLakeRequestConditions sourceConditions,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            using (Pipeline.BeginLoggingScope(nameof(PathClient)))
+            {
+                Pipeline.LogMethodEnter(
+                    nameof(PathClient),
+                    message:
+                    $"{nameof(Uri)}: {Uri}\n" +
+                    $"{nameof(destinationPath)}: {destinationPath}\n" +
+                    $"{nameof(destConditions)}: {destConditions}\n" +
+                    $"{nameof(sourceConditions)}: {sourceConditions}");
+                try
+                {
+                    DataLakeUriBuilder uriBuilder = new DataLakeUriBuilder(_dfsUri);
+                    string renameSource = "/" + uriBuilder.FileSystemName + "/" + uriBuilder.DirectoryOrFilePath;
```

Comment:
Is this just the Path minus QueryString?

Reply:
it's the file system name + the path.  Also, it has to start with '/'.